orderer/consensus/follower/follower_chain.go

```diff
@@ -199,31 +235,69 @@ func (c *Chain) StatusReport() (types.ClusterRelation, types.Status) {
 	return clusterRelation, status
 }
 
+func (c *Chain) IsRunning() bool {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	if c.started {
+		select {
+		case <-c.doneChan:
+			return false
+		default:
+			return true
+		}
+	}
+
+	return false
+}
+
 func (c *Chain) run() {
 	c.logger.Debug("The follower.Chain puller goroutine is starting")
-	pullRetryTicker := time.NewTicker(500 * time.Millisecond) // TODO move constant to options
 
 	err := c.pull()
 
 runLoop:
 	for err != nil {
-		c.logger.Debugf("Pull failed, going to try again in %v; error: %s", pullRetryTicker, err)
+		c.logger.Debugf("Pull failed, going to try again in %v; error: %s", c.retryInterval, err)
 
 		select {
 		case <-c.stopChan:
 			c.logger.Debug("Received a stop signal")
 			break runLoop
 
-		case <-pullRetryTicker.C:
+		case <-time.After(c.retryInterval):
 			err = c.pull()
 		}
+
+		if err != nil {
+			c.increaseRetryInterval()
+		}
 	}
 
-	pullRetryTicker.Stop()
 	close(c.doneChan)
 	c.logger.Debug("The follower.Chain puller goroutine is exiting")
 }
 
+func (c *Chain) increaseRetryInterval() {
+	if c.retryInterval == c.options.PullRetryMaxInterval {
```

Comment:
perhaps it should be >= ?

Reply:
If it is `>` I want to reset it to the max... only if it had reached the max in a previous round I want to return early.